Pager/Adapters/CountAdapter.php

```diff
@@ -26,11 +26,15 @@ class CountAdapter implements PagerAdapterInterface
     /**
      * Constructor.
      *
-     * @param int $count
+     * @param array|\Countable|int $value
      */
-    public function __construct($count)
+    public function __construct($value)
     {
-        $this->count = $count;
+        if (is_array($value)) {
```

Comment:
There also check for Countable interface not just for array as doc comment says it is valid argument.